### ISSUE TITLE
Component is not being re-rendered on state change

### ISSUE THREAD

**Issue by ri8ika** [avkonst/hookstate#225]
I'm setting errors state on catch block but this is not causing component re-render. 
```
schema.validate(validation).catch(err => {
  errors.merge({name: "error"})
})
```

I expected the component re-render but it didn't. To cause re-render, currently I'm doing a trick by getting state first:

```
schema.validate(validation).catch(err => {
  errors.get() // now, this cause re-render
  errors.merge({name: "error"})
})
```

Why so? I also tried set method instead of merge but still not getting expected result.
Note: The issue persist only for the first time. On next time catch, the component re-renders as expected.

**Comment by avkonst** [comment]
Hi. Thanks for the bug report. It is interesting. Would you be able to submit complete reproducer in a separate repository or in a sandbox? It is currently impossible to know what you are doing exactly and how to diagnose the problem.. Thanks.

**Comment by ri8ika** [comment]
Okay, will setup a sandbox.

**Comment by ri8ika** [comment]
Can you please try this?
```
import React, { none, useState } from '@hookstate/core'

export default function App() {
  const state = useState({})
  const errors = useState({})
  const handleSubmit = (e) => {
    // Without filling the input, Click on Add
    // Error appears only after second time button click

    // Uncomment to cause re-render
    // errors.get()
    errors.merge({Name: "Error"})
    e.preventDefault()
  }

  const onInputChange = (e) => {
    errors.merge({[e.target.name]: none})
    state.merge({[e.target.name]: e.target.value})
  }
  return (
    <div className="App">
      <form autoComplete="off" onSubmit={handleSubmit}>
          <input type="text" placeholder="Name" value={state.nested("Name").value} name="Name" onChange={onInputChange} />
          {errors.nested("Name").value && <p className="danger">{errors.nested("Name").value}</p>}
          <button type="submit">Add</button>
        </form>
    </div>
  );
}
```

**Comment by avkonst** [comment]
Thanks. Yes, I can reproduce. It is serious bug, but nested function is rarely used so not a big exposure. I will fix it.

Here is a workaround for now: use `errors["Name"].value` or `errors.Name.value` instead of `errors.nested("Name").value`.

**Comment by avkonst** [comment]
and it is only an issue if the field is undefined on the initial render

**Comment by avkonst** [comment]
fixed in 3.0.10
